File: gitdot-core/src/dto/domain/migration/github/create_github_installation.rs
use uuid::Uuid;

use super::{GitHubAppInstallAction, GitHubInstallationResponse};

#[derive(Debug, Clone)]
pub struct CreateGitHubInstallationRequest {
    pub installation_id: i64,
    pub owner_id: Uuid,
    pub state: String,
    pub code: String,
}

impl CreateGitHubInstallationRequest {
    pub fn new(installation_id: i64, owner_id: Uuid, state: String, code: String) -> Self {
        Self {
            installation_id,
            owner_id,
            state,
            code,
        }
    }
}

#[derive(Debug, Clone)]
pub struct CreateGitHubInstallationResponse {
    pub installation: GitHubInstallationResponse,
    pub action: GitHubAppInstallAction,
}

File: gitdot-web/packages/gitdot-api/src/endpoint/migration/github/create-github-installation.ts
import { z } from "zod";
import { GitHubInstallationResource } from "../../../resource";
import type { Endpoint } from "../../endpoint";

export const CreateGitHubInstallationRequest = z.object({
  state: z.string(),
  code: z.string(),
});
export type CreateGitHubInstallationRequest = z.infer<
  typeof CreateGitHubInstallationRequest
>;

export const CreateGitHubInstallationResponse = z.object({
  installation: GitHubInstallationResource,
  action: z.enum(["migration", "onboarding"]),
});
export type CreateGitHubInstallationResponse = z.infer<
  typeof CreateGitHubInstallationResponse
>;

export const CreateGitHubInstallation = {
  path: "/migration/github/{installation_id}",
  method: "POST",
  request: CreateGitHubInstallationRequest,
  response: CreateGitHubInstallationResponse,
} as const satisfies Endpoint;
export type CreateGitHubInstallation = typeof CreateGitHubInstallation;
